CMakeLists.txt

```diff
@@ -20,7 +20,20 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0069 NEW)  # LTO support
 cmake_policy(SET CMP0091 NEW)  # MSVC runtime support
 
-project(astcenc VERSION 3.0.0)
+project(astcencoder VERSION 3.0.0)
+
+option(UNIVERSAL_BUILD "Build multi-architecture universal binary [only supported on MacOS]" OFF)
+
+if (CMAKE_OSX_ARCHITECTURES)
+    message(STATUS "Building universal binary for ${CMAKE_OSX_ARCHITECTURES}")
+    set(UNIVERSAL_BUILD ON)
+endif()
+
+if (${UNIVERSAL_BUILD})
```

Comment:
Do we need UNIVERSAL_BUILD as a variable? Can't we just infer it from the fact that CMAKE_OSX_ARCHITECTURES contains more than one architecture?

Reply:
The current code no longer lets us build different SIMD binaries for macOS. It relies on CMAKE_OSX_ARCHITECTURES to select the targets, and the output is always a universal binary called `astcenc`. This is a regression for building distributable binaries; we need the ability to build all three x64 variants for binary releases (with different file names, as per the non-macOS behavior).

Reply:
> `cmake -H. -Bbuild -DISA_AVX2=ON -DISA_SSE2=ON` should still work

Ack. Was begin dense last night =)

For `UNIVERSAL_BUILD` my main question was really whether we need it to be exposed as a user-visible `option()`. The user must set `CMAKE_OSX_ARCHITECTURES` for macOS which always implies `UNIVERSAL_BUILD`, so as a user-visible option having both seems redundant. Just turn `UNIVERSAL_BUILD` into an internal variable using `set()`?

Reply:
User option removed in https://github.com/ARM-software/astc-encoder/pull/266/commits/b94c1f99323bc2d0a9bf63a6819ed27303671764